### PR TITLE
Fix bad pssa settings path crashes PSES

### PR DESCRIPTION
**Pull request by rkeithhill**
Crash happens if user tries to set path to something like `${env:HOME}\pssettings.psd1` which looks like it might be valid but isn't because the variable doesn't get expanded.

It seems the GH diff view has gotten dumber or something. This is the diff I see in VS:

image

### CODE REVIEW

File: src/PowerShellEditorServices.Protocol/Server/LanguageServerSettings.cs [hunk 1]
```diff
@@ -66,31 +66,42 @@ public ScriptAnalysisSettings()
...
                 }
+                catch (Exception ex) when (ex is NotSupportedException)
+                {
+                    // Invalid chars in path like ${env:HOME} can cause Path.GetFullPath() to throw, catch such errors here
+                    logger.WriteException(
+                        $"Invalid Script Analyzer settings path - '{settingsPath}'.",
+                        ex);
+                }
 
                 this.SettingsPath = settingsPath;
```
Comment (tylerl0706) [inline src/PowerShellEditorServices.Protocol/Server/LanguageServerSettings.cs]: Do we still want to set `this.SettingsPath` to `settingsPath` even if `settingsPath` is an Invalid Script Analyzer settings path?

Comment (rkeithhill) [reply src/PowerShellEditorServices.Protocol/Server/LanguageServerSettings.cs]: Doh!  No.  Thanks for catching this.  I'll fix it later tonight.

Comment (rkeithhill) [reply src/PowerShellEditorServices.Protocol/Server/LanguageServerSettings.cs]: Fixed.

**Comment by rkeithhill** [comment]
> A smaller try/catch helps to understand what is actually throwing the errors we are catching IMO

While that is true, my concern is that any future changes to this method skip providing such error recovery. By having an encompassing try/catch we "handle" PSSA settings path errors by logging an error instead of crashing the lang server.   The line we have to walk is to not hide "developer" errors which is why I don't like a naked `catch (Exception ex)` in these cases and prefer to use a filter.  In fact, I just noticed I'm missing `PathTooLongException`. I'll add that and update the PR.  

FWIW I fixed a number of these bugs lately (particularly PathTooLongExc) where either a user supplied setting value OR an environmental factor like a path that is too long where the fix is to do just this - catch these exceptions (where we can't do anything useful but shouldn't crash - if possible).

**Comment by tylerl0706** [comment]
Fair point! I agree with you, then. I sign off. 👍

**Pull request merged**